src/RouteDescriber/RouteArgumentDescriber/SymfonyMapQueryParameterDescriber.php

```diff
@@ -141,4 +141,35 @@ private function describeValidateFilter(?int $filter, int $flags, array $options
 
         return [];
     }
+
+    private function getEcmaRegexpFromPCRE(string $pcreRegex): string
+    {
+        // Check if PCRE regex has delimiters
+        if (!preg_match('/^(.)(.*)\1([a-zA-Z]*)$/s', $pcreRegex, $matches)) {
+            throw new \InvalidArgumentException('Invalid PCRE regex format. Missing delimiters.');
+        }
+
+        [$fullMatch, $delimiter, $pattern, $flags] = $matches;
+
+        // Check for unsupported PCRE specific constructs
+        $unsupportedFeatures = [
+            '\A', // Start of string (use ^ in JavaScript)
+            '\z', // End of string (use $ in JavaScript)
+            '\Z', // End of string before newline (not supported in JavaScript)
+            '\R', // Any Unicode newline sequence (not supported in JavaScript)
+            '\K', // Resets the start of the current match (not supported in JavaScript)
+        ];
+
+        foreach ($unsupportedFeatures as $feature) {
+            if (str_contains($pattern, $feature)) {
+                throw new \InvalidArgumentException("Unsupported PCRE feature found: {$feature}");
```

Comment:
It might not always be possible for users of this bundle to update the regexp for php's `filter_var` because this might result in different behaviour.

```suggestion
                return $pcreRegex;
```